app/views/manageBusinesses/add/IncomeSourceAddedObligations.scala.html

```diff
@@ -97,105 +84,166 @@ <h1 class="govuk-panel__title">
     }
 }
 
+@getManageUrl = @{
+    controllers.manageBusinesses.routes.ManageYourBusinessesController.show(isAgent).url
+}
+
+@getNextUpdatesUrl = @{
+    if(isAgent) {
+        controllers.routes.NextUpdatesController.showAgent.url
+    } else {
+        controllers.routes.NextUpdatesController.show().url
+    }
+}
+
+@getReportingFrequencyUrl = @{
+    // TODO the below link will need to change to the new entry point for the opt in/out journeys once the page is made
+    if(isAgent) {
+      controllers.routes.NextUpdatesController.showAgent.url
+    } else {
+      controllers.routes.NextUpdatesController.show().url
+    }
+}
+
+@getSoftwareUrl = @{
+    "https://www.gov.uk/guidance/find-software-thats-compatible-with-making-tax-digital-for-income-tax"
+}
+@getOverdueObligationsMessage(overdueObligationsComponents: OverdueObligationsMessageComponents) = @{
+    messages(s"business-added.${overdueObligationsComponents.messageKey}", overdueObligationsComponents.args: _*)
+}
 
-@mainTemplate(pageTitle = getPageTitle,
+  @getQuarterlyOrHybridList = @{
+    val quarterlyBullet: List[Html] = sources.getFirstUpcomingQuarterlyDate(currentDate) match {
+      case Some(quarterlyUpdate) =>
+        val quarterlyUpdateTaxYear = sources.getQuarterlyObligationTaxYear(quarterlyUpdate)
+        List(Html(
+        s"""<li id="quarterly-list">
+          ${getMessage("next-quarterly-obligation-text", (quarterlyUpdateTaxYear - 1).toString, quarterlyUpdateTaxYear.toString, quarterlyUpdate.inboundCorrespondenceDue.toLongDate, quarterlyUpdate.inboundCorrespondenceFrom.toLongDate, quarterlyUpdate.inboundCorrespondenceTo.toLongDate)}
+        </li>"""
+        ))
+      case None => Nil
+    }
+
+    val hybridOrFinalDecBullet: List[Html] = sources.getFinalDeclarationDate(currentDate) match {
+      case Some(finalDecDate) =>
+        List(Html(
+        s"""<li id="obligations-list">
+        ${getMessage("final-dec-text", finalDecDate.inboundCorrespondenceFrom.getYear.toString, finalDecDate.inboundCorrespondenceTo.getYear.toString, finalDecDate.inboundCorrespondenceDue.toLongDate)}
+        </li>"""
+        ))
+      case None => Nil
+    }
+
+    quarterlyBullet ++ hybridOrFinalDecBullet
+  }
+
+  @mainTemplate(pageTitle = getPageTitle,
     isAgent = isAgent,
     btaNavPartial = user.btaNavPartial) {
 
-    @formWithCSRF(action = getAddIncomeSourcesCall) {
+      <div class="govuk-panel govuk-panel--confirmation" id="banner">
+        @getBannerContent
+      </div>
+
+      @if(sources.getOverdueObligationsMessageComponents(currentDate, isBusinessHistoric).messageKey.nonEmpty) {
```

Comment:
Could extract 'sources.getOverdueObligationsMessageComponents(currentDate, isBusinessHistoric).messageKey'
for readability

Reply:
That's a good shout, also extracted something else a couple lines down too.

Reply:
nice one